libr/parse/p/parse_ppc_pseudo.c

```diff
@@ -10,231 +10,1460 @@
+                            argv[ops[i].max_operands] = argv[ops[i].max_operands - 1];
+                            if (ops[i].max_operands == 3) {
+                                argv[2] = argv[1];
+                            }
+                            tmp[0] = 'c';
+                            tmp[1] = 'r';
+                            tmp[2] = '0';
+                            tmp[3] = '\0';
+                            argv[1] = tmp;
+                        }
+                        int letter = ops[i].str[j] - '@';
+                        const char *w = argv[letter];
+                        if (letter == 4 && i == 27) {
+                            w = inv_mask64(argv[4], argv[3]);
+                        } else if (letter == 4 && i >= 28 && i <= 29) {
+                            w = cmask64(w, "63");
+                        } else if (letter == 4 && i >= 30 && i <= 31) {
+                            w = cmask64("0", w);
+                        } else if (letter == 4 && i >= 30 && i <= 31) {
+                            w = cmask64("0", w);
+                        } else if (letter == 4 && i == 32) {
+                            w = inv_mask64(argv[4], argv[3]);
+                        } else if (letter == 4 && i >= 33 && i <= 35) {
+                            w = cmask32(argv[4], argv[5]);
+                        } else if (letter == 1 && i >= 36 && i <= 43) {
+                            int to = atoi(w);
+                            switch(to) {
+                            case 4:
+                                w = "==";
+                            case 1:
+                            case 8:
+                                w = ">";
+                            case 5:
+                            case 12:
+                                w = ">=";
+                            case 2:
+                            case 16:
+                                w = "<";
+                            case 6:
+                            case 20:
+                                w = "<=";
+                            case 24:
+                                w = "!=";
+                            case 31:
+                            // If no parameters t[dw][i] 32, 0, 0 just TRAP
+                                w = "==";
+                            }
+                            w = cmask64("0", w);
+                        }
+                        if (w != NULL) {
+                            strcpy (newstr + k, w);
+                            k += strlen(w)-1;
+                        }
+                    } else newstr[k] = ops[i].str[j];
+                }
+                newstr[k]='\0';
+            }
+            return true;
+        }
+    }
+
+    /* TODO: this is slow */
+    if (newstr != NULL) {
+        newstr[0] = '\0';
+        for (i = 0; i < argc; i++) {
+            strcat (newstr, argv[i]);
+            strcat (newstr, (i == 0 || i == argc - 1) ? " ":", ");
```

Comment:
how do i check the length of that? the string pointer comes from the `parse(RParse *p, const char *data, char *str)` caller.